source/common/router/router.h

```diff
@@ -371,19 +369,25 @@ class Filter : Logger::Loggable<Logger::Id::router>,
                                          Upstream::ResourcePriority priority) PURE;
   Http::ConnectionPool::Instance* getConnPool();
   void maybeDoShadowing();
+  bool maybeRetryReset(const Http::StreamResetReason reset_reason);
+  void onPerTryTimeout();
   void onRequestComplete();
   void onResponseTimeout();
   void onUpstream100ContinueHeaders(Http::HeaderMapPtr&& headers);
+  // Handle an "aborted" upstream request, meaning we didn't see response
+  // headers (e.g. due to a reset). Handles recording stats and responding
+  // downstream if appropriate.
+  void onUpstreamAbort(const Http::Code code, const StreamInfo::ResponseFlag response_flag, const std::string& body, bool dropped);
```

Comment:
I think the originally comment got lost: I'd remove the const for the non-reference types here and on `maybeRetryReset`